apps/web/src/components/NotificationDialog/NotificationDialog.tsx

```diff
@@ -44,9 +45,125 @@ const ellipsis = () => (
   />
 );
 
-const NotificationDialog = () => {
+const DialogPortalWithPresence = ({
+  executeQuery,
+}: {
+  executeQuery: UseQueryExecute;
+}) => {
   const intl = useIntl();
   const paths = useRoutes();
+  const [isPresent] = usePresence();
+  const [render, setRender] = useState<boolean>(isPresent);
+  useEffect(() => {
+    if (isPresent) {
+      setRender(true);
+    } else {
+      setTimeout(() => setRender(false), 200); // let animation complete before removal
```

Comment:
I noticed this because it came up in my PR, too.
https://github.com/GCTC-NTGC/gc-digital-talent/pull/10824#discussion_r1662609964

You should probably get the return from setTimeout and clean it up.

Reply:
Like so https://github.com/GCTC-NTGC/gc-digital-talent/pull/10843/commits/8638d9427131be3ea2a7e316a67deb710426db79 ?

Reply:
Nice.